lib/ttcp

```diff
@@ -191,21 +214,144 @@ __ttcp::spin () {
     _TTCP_SPIN_PID=$!
 }
 
-__ttcp::check_env () {
+__ttcp::generate_credentials () {
+    TTCP_ID_CLIP=${TTCP_ID_CLIP:-"$(echo "${TTCP_ID}${TTCP_SALT1}" | __ttcp::hash)"}
+    TTCP_PASSWORD_CLIP=${TTCP_PASSWORD_CLIP:-"$(echo "${TTCP_PASSWORD}${TTCP_SALT1}" | __ttcp::hash)"}
+    TTCP_ID_TRANS=${TTCP_ID_TRANS:-"$(echo "${TTCP_ID}${TTCP_SALT2}" | __ttcp::hash)"}
+    TTCP_PASSWORD_TRANS=${TTCP_PASSWORD_TRANS:-"$(echo "${TTCP_PASSWORD}${TTCP_ID}${TTCP_SALT2}" | __ttcp::hash)"}
+}
+
+__ttcp::init () {
+    # Stop immediately in case of any errors.
+    # Because this function may make strange changes.
+    set -e
+
+    # Get a filename calling the function.
+    # http://stackoverflow.com/a/192319/
+    local _file="${0##*/}"
+    # Remove file's extention
+    local _cmd="${_file%.*}"
+
+    local _given_id=""
+    local _given_password1=""
+    local _given_password2=""
+
+    # If there is any standard input, discard it.
+    # If users try to use ttcopy with stdin for the first time.
+    if [ ! -t 0 ]; then
+        echo "Standard input is discarded because credentials are not set yet." >&2
+        echo "Please try again after setting ID/Password." >&2
+        echo >&2
+        cat /dev/stdin > /dev/null
+    fi
+
+    echo "Set default ID/Password." >&2
+    # Following messages are similar to ssh-keygen's one.
+
+    printf "Enter ID for ttcopy/ttpaste:" >&2
+    read _given_id < ${INPUT_DEVICE}
+
+    printf "Enter password for ttcopy/ttpaste:" >&2
+    read -s _given_password1 < ${INPUT_DEVICE}
+    echo >&2
+
+    printf "Enter same password again:" >&2
+    read -s _given_password2 < ${INPUT_DEVICE}
+    echo >&2
+
+    # Check passwords
+    if [[ "$_given_password1" !=  "$_given_password2" ]]; then
+        echo "Passwords do not match. Try again." >&2
+        exit $_TTCP_EUNMATCH
+    fi
+
+    # Try to create a directory to store credentials in any case.
+    mkdir -p "$_TTCP_CONFIG_DIR"
+    # Force remove in advance, because the file may not have writable attribute.
+    rm -f $_TTCP_CONFIG_FILE
+    # Create blank file.
+    touch $_TTCP_CONFIG_FILE
+
+    TTCP_ID="$_given_id"
+    TTCP_PASSWORD="$_given_password1"
+    __ttcp::generate_credentials
+
+    echo "TTCP_ID=$TTCP_ID" > $_TTCP_CONFIG_FILE
+    echo "TTCP_ID_CLIP=$TTCP_ID_CLIP" >> $_TTCP_CONFIG_FILE
+    echo "TTCP_PASSWORD_CLIP=$TTCP_PASSWORD_CLIP" >> $_TTCP_CONFIG_FILE
+    echo "TTCP_ID_TRANS=$TTCP_ID_TRANS" >> $_TTCP_CONFIG_FILE
+    echo "TTCP_PASSWORD_TRANS=$TTCP_PASSWORD_TRANS" >> $_TTCP_CONFIG_FILE
+
+    # Prohibit users (except for root & current user) from seeing the file.
+    chmod 400 $_TTCP_CONFIG_FILE
```

Comment:
I believe, there is no chance for users to edit this file **basically**.
But it might be too much restriction. (Even the `id_rsa` is 600).
Skilled user may edit it, therefore I will fix this part.